effekt/shared/src/main/scala/effekt/source/Tree.scala

```diff
@@ -209,6 +250,73 @@ enum Param extends Definition {
 }
 export Param.*
 
+/**
+ * A `List` (usually of `Tree`s) with a `span` that spans all elements of the list
+ */
+case class Many[T](list: List[T], span: Span) {
+  def unspan: List[T] = list
+  def map[B](f: T => B): Many[B] =
+    Many(list.map(f),span)
+
+  def unzip [A1, A2](implicit asPair: T => (A1, A2)): (Many[A1], Many[A2]) = {
+    val (list1 : List[A1], list2 : List[A2]) = list.unzip(asPair)
+    (Many(list1, span), Many(list2, span))
+  }
+
+  def collect[B](pf: PartialFunction[T, B]): Many[B] =
+    Many(list.collect(pf), span)
+
+  def zip[B](that: IterableOnce[B]): Many[(T, B)] = Many(list.zip(that), span)
+
+  def flatMap[B](f: T => IterableOnce[B]): Many[B] =
+    Many(list.flatMap(f), span)
+
+  export list.{foreach, toSet, isEmpty, nonEmpty, mkString, size, length, init, last}
+}
+
+object Many {
+   def empty[T](span: Span) = Many[T](Nil, span)
+}
+
+/**
+ * An `Option` with a `span`
+ *
+ * If the `Option` is `Some`, the `span` is the span of the value.
+ * If the `Option` is `None`, the `span` is an empty span pointing to where the optional value would be expected.
+ */
+case class Maybe[T](option: Option[T], span: Span){
+  def unspan: Option[T] = option
```

Comment:
Similar comment as in Many: why both the `.option` and the `.unspan` at the same time?